graspy/align/orthogonal_procrustes.py

```diff
@@ -0,0 +1,200 @@
+    ):
+        # check type
+        if not isinstance(align_type, str):
+            msg = "align_type must be a string, not {}".format(type(align_type))
+            raise TypeError(msg)
+        align_types_supported = [
+            "orthogonal",
+            "scaling-orthogonal",
+            "diagonal-orthogonal",
+        ]
+        if align_type not in align_types_supported:
+            msg = "supported types are {}".format(align_types_supported)
+            raise ValueError(msg)
+        if type(scale) is not float:
+            msg = "scale must be a float, not {}".format(type(scale))
+            raise TypeError(msg)
+        # Value checking
+        if scale <= 0:
+            msg = "{} is an invalud value of the optimal transport eps, must be postitive".format(
+                scale
+            )
+            raise ValueError(msg)
+
+        super().__init__(freeze_Y=freeze_Y)
+        self.align_type = align_type
+        self.scale = scale
+
+    def fit(self, X, Y):
+        """
+        Uses the two datasets to learn matrices Q_X and Q_Y.
+        In regular orthogonal procrustes Q_X is a solution to the orthogonal
+        procrustes problem, and Q_Y is the identity matrix.
+
+        Parameters
+        ----------
+        X: np.ndarray, shape (n, d)
+            First dataset of vectors. These vectors need to have same number of
+            dimensions as ones in Y, but the number of vectors can differ.
+
+        Y: np.ndarray, shape (m, d)
+            Second dataset of vectors. These vectors need to have same number
+            of dimensions as ones in X, but the number of vectors can differ.
+
+        Returns
+        -------
+        self: returns an instance of self
+        """
+
+        # check for numpy-ness, 2d-ness and finite-ness
+        if not isinstance(X, np.ndarray):
+            msg = f"first dataset is a {type(X)}, not an np.ndarray! "
+            raise TypeError(msg)
+        if not isinstance(Y, np.ndarray):
+            msg = f"first dataset is a {type(Y)}, not an np.ndarray! "
+            raise TypeError(msg)
+        X = check_array(X, accept_sparse=True, copy=True)
+        Y = check_array(Y, accept_sparse=True, copy=True)
+
+        # check for equal components and number of entries
+        if X.shape[1] != Y.shape[1]:
+            msg = "two datasets have different number of components!"
+            raise ValueError(msg)
+        _, d = X.shape
+        if X.shape[0] != Y.shape[0]:
+            msg = (
+                "two datasets have different number of entries! "
+                "OrthogonalProcrustes assumes that entries of the two "
+                "datasets are matched. consider using SeedlessProcrustes "
+                "instead."
+            )
+            raise ValueError(msg)
+
+        _, d = X.shape
+        if self.align_type == "orthogonal":
+            D_X = np.eye(d)
+            D_Y = np.eye(d)
+        if self.align_type == "scaling-orthogonal":
```

Comment:
@bdpedigo in other words, pls explein why this isnt what im supposed to be doing.